main.tf

```diff
@@ -468,111 +567,3 @@ resource "aws_eks_identity_provider_config" "this" {
 
   tags = var.tags
 }
-
-################################################################################
-# aws-auth configmap
-################################################################################
-
-locals {
-  node_iam_role_arns_non_windows = distinct(
-    compact(
-      concat(
-        [for group in module.eks_managed_node_group : group.iam_role_arn if group.platform != "windows"],
-        [for group in module.self_managed_node_group : group.iam_role_arn if group.platform != "windows"],
-        var.aws_auth_node_iam_role_arns_non_windows,
-      )
-    )
-  )
-
-  node_iam_role_arns_windows = distinct(
-    compact(
-      concat(
-        [for group in module.eks_managed_node_group : group.iam_role_arn if group.platform == "windows"],
-        [for group in module.self_managed_node_group : group.iam_role_arn if group.platform == "windows"],
-        var.aws_auth_node_iam_role_arns_windows,
-      )
-    )
-  )
-
-  fargate_profile_pod_execution_role_arns = distinct(
-    compact(
-      concat(
-        [for group in module.fargate_profile : group.fargate_profile_pod_execution_role_arn],
-        var.aws_auth_fargate_profile_pod_execution_role_arns,
-      )
-    )
-  )
-
-  aws_auth_configmap_data = {
-    mapRoles = yamlencode(concat(
-      [for role_arn in local.node_iam_role_arns_non_windows : {
-        rolearn  = role_arn
-        username = "system:node:{{EC2PrivateDNSName}}"
-        groups = [
-          "system:bootstrappers",
-          "system:nodes",
-        ]
-        }
-      ],
-      [for role_arn in local.node_iam_role_arns_windows : {
-        rolearn  = role_arn
-        username = "system:node:{{EC2PrivateDNSName}}"
-        groups = [
-          "eks:kube-proxy-windows",
-          "system:bootstrappers",
-          "system:nodes",
-        ]
-        }
-      ],
-      # Fargate profile
-      [for role_arn in local.fargate_profile_pod_execution_role_arns : {
-        rolearn  = role_arn
-        username = "system:node:{{SessionName}}"
-        groups = [
-          "system:bootstrappers",
-          "system:nodes",
-          "system:node-proxier",
-        ]
-        }
-      ],
-      var.aws_auth_roles
-    ))
-    mapUsers    = yamlencode(var.aws_auth_users)
-    mapAccounts = yamlencode(var.aws_auth_accounts)
-  }
-}
-
-resource "kubernetes_config_map" "aws_auth" {
```

Comment:
any reason to not use `moved` blocks here so folks have an easier time upgrading the new major version release?

Reply:
yes - what am I moving it to? 😉 

unfortunately, move blocks are good for refactoring type changes that are all kept within the module, but theres a lot of rough edges around trying to move things that have for_each, moving things in/out of a module, etc. 

we will have guidance around upgrading - fortunately, nothing is control plane impacting, but there will be some one time handling of changing from `aws-auth` to cluster access entries. TBD what those options will be

Reply:
The upgrade path recommended by Amazon docs is to go with both configMap and api and only then move to api only.

Reply:
correct - we are defaulting the access entry to `API_AND_CONFIG_MAP` on this change and then we'll be changing that default to `API` only on the next release in the future. Its a one way op so, this is the safest middle group while folks migrate